src/devices/Bh1745/Bh1745.cs

```diff
@@ -0,0 +1,394 @@
+            {
+                if (!Enum.IsDefined(typeof(InterruptSource), value))
+                    throw new ArgumentOutOfRangeException();
+
+                var intSource = Read8BitsFromRegister((byte)Register.INTERRUPT);
+                intSource = (byte)((intSource & (byte)~Mask.INT_SOURCE) | (byte)value << 2);
+
+                Write8BitsToRegister((byte)Register.INTERRUPT, intSource);
+                _interruptSource = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets whether the interrupt pin is enabled.
+        /// </summary>
+        public bool InterruptIsEnabled
+        {
+            get => _interruptIsEnabled;
+            set
+            {
+                var intPin = Read8BitsFromRegister((byte)Register.INTERRUPT);
+                intPin = (byte)((intPin & (byte)~Mask.INT_ENABLE) | Convert.ToByte(value));
+
+
+                Write8BitsToRegister((byte)Register.INTERRUPT, intPin);
+                _interruptIsEnabled = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the persistence function of the interrupt.
+        /// </summary>
+        public InterruptPersistence InterruptPersistence
+        {
+            get => _interruptPersistence;
+            set
+            {
+                if (!Enum.IsDefined(typeof(InterruptPersistence), value))
+                    throw new ArgumentOutOfRangeException();
+
+                var intPersistence = Read8BitsFromRegister((byte)Register.PERSISTENCE);
+                intPersistence = (byte)((intPersistence & (byte)~Mask.PERSISTENCE) | (byte)value);
+
+                Write8BitsToRegister((byte)Register.PERSISTENCE, intPersistence);
+                _interruptPersistence = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the lower interrupt threshold.
+        /// </summary>
+        public ushort LowerInterruptThreshold
+        {
+            get => _lowerInterruptThreshold;
+            set
+            {
+                WriteShortToRegister((byte)Register.TL, value);
+                _lowerInterruptThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the higher interrupt threshold.
+        /// </summary>
+        public ushort HigherInterruptThreshold
+        {
+            get => _higherInterruptThreshold;
+            set
+            {
+                WriteShortToRegister((byte)Register.TH, value);
+                _higherInterruptThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the channel compensation multipliers which are used to compensate the measurements.
+        /// </summary>
+        public ChannelCompensationMultipliers ChannelCompensationMultipliers { get; set; }
+
+        /// <summary>
+        /// Resets the device to the default configuration.
+        /// On reset the sensor goes to power down mode.
+        /// </summary>
+        public void Reset()
```

Comment:
Should you also update any of the private fields too to reflect the state?

Reply:
Oh yeah good catch forgot about that when I changed from actual device polling to caching through fields.